### PR TITLE
Redo the Pass Manager

### PR DESCRIPTION
**Pull request by CodaFi**
This thing was never correct.  We mixed up Function passes and Module passes, and if you were unlucky enough to schedule both into the function pass manager then LLVM would probably crash.

Provide the interface we were shooting for here anyways in the form of a Swift-like pass pipeliner.  You can freely mix module and function-level passes in the pipeliner, and we will schedule them the way LLVM would anyways (group the function passes, then run the module passes).  There's bindings for LTO passes that never made it into a header that we could also wrap with this infrastructure.

### CODE REVIEW

File: Sources/LLVM/PassPipeliner.swift [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+  var isModulePass: Bool {
+    switch self {
+    case .stripSymbols,
+         .stripDeadPrototypes,
+         .constantMerge,
+         .globalOptimizer,
+         .globalDCE,
+         .deadArgElimination,
+         .ipConstantPropagation,
+         .ipscc:
```
Comment (matthewseaman) [inline Sources/LLVM/PassPipeliner.swift]: I believe at least .functionAttrs should be here too.

File: Sources/LLVM/PassManager.swift [hunk 2]
```diff
@@ -106,8 +106,6 @@ public enum FunctionPass {
   case scalarReplAggregates
   /// Replace aggregates or pieces of aggregates with scalar SSA values.
   case scalarReplAggregatesSSA
-  /// Tries to inline the fast path of library calls such as sqrt.
-  case simplifyLibCalls
```
Comment (matthewseaman) [inline Sources/LLVM/PassManager.swift:110]: Why is this removed?

**Review by matthewseaman — COMMENTED** [review]
Looking great!

File: Sources/LLVM/PassManager.swift (hunk 2 above)
Comment (CodaFi) [reply Sources/LLVM/PassManager.swift:110]: The corresponding add pass function is a no-op as the pass has been removed.

**Comment by CodaFi** [comment]
⛵️
